syncr_backend/network/handle_frontend.py

```diff
@@ -474,19 +492,41 @@ def handle_input_name(
     :param conn: socket.accept() connection
     :return: None
     """
-    if request['drop_name'] is None:
-        response = {
-            'status': 'error',
-            'error': ERR_INVINPUT,
-        }
+    # This code assumes that the user has already
+    # created a folder to initialize as a drop.
+    #
+    # First, we check to see if tracker already contains drop id
+    # If so, do nothing, send message failure back
+    #
+    # Else, initialize directory with 'drop_name' as the name of the drop
+
+    # TODO: Allow user to select directory location from UI (on frontend).
+
+    directory = request['directory']
+    drop_name = os.path.basename(directory)
+
+    status = 'error'
+    result = 'failure'
+
+    # TODO: Change 'False' to check if drop id already exists in tracker
+    if False:
+        message = 'A drop already exists with the given drop name'
     else:
-        # TODO: backend logic to create a drop.
-        # TODO: Test if given drop_name is valid.
-        response = {
-            'status': 'ok',
-            'result': 'success',
-            'message': 'Drop ' + request['drop_name'] + ' created',
-        }
+
+        try:
+            initialize_drop(directory)
+        except RuntimeError:
+            message = 'Error in initializing drop.'
+        else:
+            status = 'ok'
+            result = 'success'
+            message = 'Drop ' + drop_name + 'created'
+
+    response = {
+        'status': status,
+        'result': result,
+        'message': message,
+    }
```

Comment:
Initializing a drop should also then display the drop on frontend. Therefore, we will want to send that drop info back as a requested_drop to frontend. Look at my pull request for a function that changes the formatting of the DropMetadata objects to just regular old dictionaries that the frontend si expecting.worst case i can just update this after everything is merged

Reply:
Wait, nvm the frontend just recalls this specific drop request later. This is fine as is